### ISSUE TITLE
References displayed in PW

### ISSUE THREAD

**Issue by DeniseSl22** [wikipathways/wikipathways.org#74]
Hi all,

Just a reminder about the references in pathways (on the PW itself, on nodes, on interactions), which are currently not visualized in the diagram. It would be nice if this could happen sometimes (several people asked me about this). Especially for biologists+educators this can be really beneficial (to check the validity of (a part of) a PW). And would be really cool if the references are not only shown, but also include a clickable link to the corresponding pubmed article :)

**Comment by DeniseSl22** [comment]
And I also noticed that the order of the references on the website is not the same as the numbers they get in the GPML.... why is that? makes it really hard to trace back something from the website to the pathway (as an example see WP1604, has references all over the place to good testcase)

**Comment by Chris-Evelo** [comment]
I am surprised that in pathway references are not clickable at all. I remember seeing them as numbers in the pathway diagram that could be clicked to show the full reference. Did we lose that mechanism at some stage?

**Comment by DeniseSl22** [comment]
I think this got lost with the new json release (just as the "search bar" within a PW?

**Comment by egonw** [comment]
I think you meant "with the new pvjs release"... actually, I think references in pvjs just never got implemented yet... it's all down to Anders again, tho we had Jacob to help out from our side for a while...

**Comment by DeniseSl22** [comment]
Ooh sorry, yes I meant new pvjs release (sometimes the abbreviations get mixed up in my head ;) ...)

Well, if only Anders is busy with this now, maybe we should get some other people involved again (either good students like Jacob, or make a new google summer of code project for this?). Would also be nice to use the projects that Jacob worked on and really integrate them in WPs.

**Comment by AlexanderPico** [comment]
Agreed. Bringing back the dispaly reference numbers and the in-pathway search bar would be good GSoC projects or entry projects for others in the team interested in JavaScript development.  My team won't be able to invest more time on pvjs for a while...
